modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/ProjectedTuple.java

```diff
@@ -45,49 +41,47 @@ public class FormatAwareProjectedTuple extends AbstractProjectedTuple {
      *         an index of field in resulting projection, and an element of the array at that index is an index of column in original
      *         tuple.
      */
-    public FormatAwareProjectedTuple(InternalTuple delegate, int[] projection) {
-        super(delegate, projection);
+    public ProjectedTuple(InternalTuple delegate, int[] projection) {
+        super((InternalTupleEx) delegate, projection);
```

Comment:
tbh I don't really like this approach with force casting. Let's try to preserve type safety

Reply:
Fixed